k2/python/host/tests/arcsort_test.py

```diff
@@ -45,9 +46,12 @@ def test_arc_sort(self):
         arc_map = k2host.IntArray1.create_array_with_size(fsa.size2)
         k2host.arc_sort(fsa, arc_map)
         expected_arc_indexes = torch.IntTensor([0, 3, 5, 6, 6, 6])
-        expected_arcs = torch.IntTensor([[0, 2, 0, 3], [0, 4, 0, 2],
-                                         [0, 1, 2, 1], [1, 3, 0, 5],
-                                         [1, 2, 1, 4], [2, 1, 0, 6]])
+        expected_arcs = torch.IntTensor([[0, 2, 0, float_to_int(3)],
```

Comment:
I guess casting type one by one element is not so efficient.

Reply:
the time would mostly be an issue in test code.  we can work out more efficient ways to do this in real code, e.g. by reinterpreting entire tensors, which I think you can do as `cast` in torch.
I want to keep this design, it is efficient for production stuff.  
What we could actually do is, have various convenience functions that manipulate these things, as in,
   combine_arcs(int_tensor with 3 cols, float_tensor with 1 col) -> int tensor with 1 col
   separate_arcs(int_tensor with 1 col) -> (int_tensor with 3 cols, float_tensor with 1 col) 
and for test:
  arcs_from_float(float_tensor with 4 cols) -> int_tensor with 4 cols, except last field which is a float in binary terms.

Reply:
Yeah, that is similar to the approach we used for old host python code https://github.com/k2-fsa/k2/pull/173#discussion_r493171104